qiita_pet/handlers/analysis_handlers.py

```diff
@@ -321,3 +325,31 @@ def get(self):
                     user.shared_analyses + user.private_analyses]
 
         self.render("show_analyses.html", analyses=analyses)
+
+
+class ResultsHandler(StaticFileHandler, BaseHandler):
+    def validate_absolute_path(self, root, absolute_path):
+        """Overrides StaticFileHandler's method to include authentication
+        """
+        u = self.current_user
+
+        accessible_filepaths = get_accessible_filepath_ids(u)
+        accessible_filepaths = {join(get_db_files_base_dir(),
+                                     filepath_id_to_rel_path(fpid))
+                                for fpid in accessible_filepaths}
+
+        for accessible_filepath in accessible_filepaths:
+            access = False
+
+            if (accessible_filepath == absolute_path):
+                access = True
+
+            elif isdir(accessible_filepath) and \
+                    absolute_path.startswith(accessible_filepath + sep):
+                access = True
+
+            if access:
+                return super(ResultsHandler, self).validate_absolute_path(
+                    root, absolute_path)
+
+        raise QiitaPetAuthorizationError(u.id, absolute_path)
```

Comment:
@antgonza can you clarify your first point? I don't understand what the issue is. By studies, do you mean analysis results? In this implementation, a regular user is capable of opening their own analysis results.

> Do you think is possible to check for a specific file path id...

I do not believe it is possible (assuming we still want to use `StaticFileHandler` or a subclass of it to serve results), for a couple reasons.

1) Not all results files are in the filepath table (e.g., the output of summarize taxa is a directory, which is in the filepath table, but the files contained within that directory are not in the filepath table, and these are the paths that are being served by the `StaticFileHandler`)
2) The static filehandler is only aware of the actual path being requested, and not the filepath ID.